xmtp_mls/migrations/2024-08-07-213816_create-private-preference-store/up.sql

```diff
@@ -0,0 +1,11 @@
+CREATE TABLE "private_preferences"(
+    -- The inbox_id the private preferences are owned by
+    "inbox_id" text NOT NULL,
+    -- Enum of the PRIVATE_PREFERENCE_TYPE (GROUP, INBOX, etc..)
+    "valueType" int NOT NULL,
+    -- Enum of PREFERENCE_STATE (ALLOWED, DENIED, etc..)
+    "state" int NOT NULL,
+    -- The value of what has a preference
+    "value" text NOT NULL,
+    PRIMARY KEY (valueType, value)
+);
```

Comment:
Want to make sure I'm reasoning about this correctly we will want a way to get the current consent state for a group or inbox id performantly so the primary key should be the value and the valueType so that you can fetch for type group and value group_id and get what the state is.

Reply:
That does feel like what we'll want at the end of the process. Very cheap to look up, has all the information a

One thing that was a struggle in the past is just making sure that we processed all the events in the correct order, so that if you allowed and then blocked the same group you'd get a consistent value in the end. What happens if you received a preferences event that you couldn't read before, but now you can? How do we rewind things so that you can rebuild the state? Maybe the right answer is to truncate this table and start from scratch in that scenario?

Reply:
does it make sense to make an index here, like we did for group id?

```
CREATE INDEX idx_value_type_value ON private_preferences(valuType, value ASC);
```

Reply:
My understanding is that the primary key made an index by default so we shouldn't need to add another index.